### PR TITLE
Added Unit Test Cases for Wrapper and Helper Files

### PR DESCRIPTION
**Pull request by Raj-Shah1**
### Summary by CodeRabbit

- Test: Added new unit tests for classes in the `com.salessparrow.api.unit.lib.salesforce.helper` and `com.salessparrow.api.unit.lib.salesforce.wrappers` packages. These tests cover methods related to Salesforce OAuth token management, HTTP request handling, and data retrieval from Salesforce.
- Refactor: Improved existing test cases by adding more assertions and simulating different scenarios, enhancing the robustness of the test suite.

### CODE REVIEW

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    // Assertions
+    assertNotNull(exception);
+    ErrorObject errorObject = exception.getErrorObject();
+    assertNotNull(errorObject);
+    assertEquals("l_s_w_sgi_gui_1", errorObject.getInternalErrorIdentifier());
+    assertEquals("bad_request", errorObject.getApiErrorIdentifier());
+
+    httpClientMockedStatic.close();
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    when(awsKms.encryptToken("new_access_token")).thenReturn("encrypted_access_token");
+
+    when(salesforceOauthTokenRepository.saveSalesforceOauthToken(any(SalesforceOauthToken.class)))
+      .thenAnswer(invocation -> invocation.getArgument(0)); // Return the argument back
+
+    String decryptedAccessToken = salesforceOauthToken.updateAndGetRefreshedAccessToken(sfOAuthToken);
+
+    assertEquals("new_access_token", decryptedAccessToken);
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java]: Not needed

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    };
+
+    // Test the method and expect a CustomException
+    assertThrows(CustomException.class, () -> salesforceRequest.makeRequest("user123", requestInterface));
+
+    // Verify that the methods were called with the expected parameters
+    verify(sfOauthTokenRepository).getSalesforceOauthTokenByExternalUserId("user123");
+    verify(salesforceOAuthToken).fetchAccessToken(sfOAuthToken);
+    verify(salesforceOAuthToken, times(1)).updateAndGetRefreshedAccessToken(sfOAuthToken);
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceRequestTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    httpClientMockedStatic.when(() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()))
+      .thenReturn(mockResponse);
+    
+    HttpResponse actualResponse = salesforceGetRefreshedAccessToken.getRefreshedAccessToken(refreshToken);
+
+    // Assertions
+    assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+
+    httpClientMockedStatic.close();
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetNoteContentTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+public class SalesforceGetNoteContentTest {
+
+  @Autowired
+  private SalesforceGetNoteContent salesforceGetNoteContent;
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetNoteContentTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetNoteContentTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  public void testGetNoteContent() {
+    String noteId = "note123";
+    String salesforceUserId = "user123";
+
+    // Mock SalesforceConstants
+    when(salesforceConstants.timeoutMillis()).thenReturn(5000);
+
+    // Mock SalesforceRequest
+    when(salesforceRequest.makeRequest(eq(salesforceUserId), any(SalesforceRequestInterface.class)))
+      .thenReturn(new HttpClient.HttpResponse(200, "Note Content", null, "text/plain"));
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetNoteContentTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    ErrorObject errorObject = exception.getErrorObject();
+    assertNotNull(errorObject);
+    assertEquals("l_s_w_sgi_gui_1", errorObject.getInternalErrorIdentifier());
+    assertEquals("bad_request", errorObject.getApiErrorIdentifier());
+
+    httpClientMockedStatic.verify(() -> HttpClient.makeGetRequest(instanceUrl + salesforceConstants.identityUrl(), headers, 10000), Mockito.times(1));
+
+    httpClientMockedStatic.close();
+  }
+}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    httpClientMockedStatic.when(() -> HttpClient.makeGetRequest(anyString(), anyMap(), anyInt()))
+      .thenReturn(mockResponse);
+    
+    HttpResponse actualResponse = salesforceGetIdentity.getUserIdentity(instanceUrl, accessToken);
+
+    // Assertions
+    assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+
+    httpClientMockedStatic.close();
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    when(awsKms.encryptToken("new_access_token")).thenReturn("encrypted_access_token");
+
+    when(salesforceOauthTokenRepository.updateSalesforceOauthToken(any(SalesforceOauthToken.class)))
+      .thenAnswer(invocation -> invocation.getArgument(0)); // Return the argument back
+
+    String decryptedAccessToken = salesforceOauthToken.updateAndGetRefreshedAccessToken(sfOAuthToken);
+
+    assertEquals("new_access_token", decryptedAccessToken);
+  }
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+		assertNotNull(errorObject);
+		assertEquals("l_s_w_sgi_gui_1", errorObject.getInternalErrorIdentifier());
+		assertEquals("bad_request", errorObject.getApiErrorIdentifier());
+
+		httpClientMockedStatic.verify(
+				() -> HttpClient.makeGetRequest(instanceUrl + salesforceConstants.identityUrl(), headers, 10000),
+				Mockito.times(1));
+
+		httpClientMockedStatic.close();
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java:98]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+		// Assertions
+		assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+
+		httpClientMockedStatic.verify(
+				() -> HttpClient.makeGetRequest(instanceUrl + salesforceConstants.identityUrl(), headers, 10000),
+				Mockito.times(1));
+
+		httpClientMockedStatic.close();
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetIdentityTest.java:63]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+		HttpResponse actualResponse = salesforceGetRefreshedAccessToken.getRefreshedAccessToken(refreshToken);
+
+		// Assertions
+		assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+		httpClientMockedStatic.verify(() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()),
+				times(1));
+
+		httpClientMockedStatic.close();
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java:54]: Where should I use assertThrows

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+			.thenReturn(mockResponse);
+
+		HttpResponse actualResponse = salesforceGetRefreshedAccessToken.getRefreshedAccessToken(refreshToken);
+
+		// Assertions
+		assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+		httpClientMockedStatic.verify(() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()),
+				times(1));
+
+		httpClientMockedStatic.close();
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetRefreshedAccessTokenTest.java:53]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+		when(awsKms.encryptToken("new_access_token")).thenReturn("encrypted_access_token");
+
+		when(salesforceOauthTokenRepository.updateSalesforceOauthToken(any(SalesforceOauthToken.class)))
+			.thenAnswer(invocation -> invocation.getArgument(0)); // Return the argument
+																	// back
+
+		String decryptedAccessToken = salesforceOauthToken.updateAndGetRefreshedAccessToken(sfOAuthToken);
+
+		assertEquals("new_access_token", decryptedAccessToken);
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java:83]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+		httpClientMockedStatic.when(() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()))
+			.thenReturn(mockResponse);
+
+		HttpResponse actualResponse = salesforceGetTokens.getTokens(code, redirectUri);
+
+		// Assertions
+		assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+
+		httpClientMockedStatic.close();
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java:55]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+		});
+
+		// Assertions
+		assertNotNull(exception);
+		ParamErrorObject paramErrorObject = exception.getParamErrorObject();
+		assertNotNull(paramErrorObject);
+		assertEquals("l_s_w_sgt_gt_1", paramErrorObject.getInternalErrorIdentifier());
+
+		httpClientMockedStatic.close();
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java]: Resolved

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+				.thenReturn(mockResponse);
+
+			HttpResponse actualResponse = salesforceGetTokens.getTokens(code, redirectUri);
+
+			// Assertions
+			assertEquals(mockResponse.getResponseBody(), actualResponse.getResponseBody());
+			httpClientMockedStatic.verify(
+					() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()), Mockito.times(1));
+		}
+
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java:56]: Not needed

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+			assertNotNull(exception);
+			ParamErrorObject paramErrorObject = exception.getParamErrorObject();
+			assertNotNull(paramErrorObject);
+			assertEquals("l_s_w_sgt_gt_1", paramErrorObject.getInternalErrorIdentifier());
+
+			httpClientMockedStatic.verify(
+					() -> HttpClient.makePostRequest(anyString(), anyMap(), anyString(), anyInt()), Mockito.times(1));
+
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/wrappers/SalesforceGetTokensTest.java:90]: Not needed

File: src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+		when(salesforceOauthTokenRepository.updateSalesforceOauthToken(any(SalesforceOauthToken.class)))
+			.thenAnswer(invocation -> invocation.getArgument(0)); // Return the argument
+																	// back
+
+		String decryptedAccessToken = salesforceOauthToken.updateAndGetRefreshedAccessToken(sfOAuthToken);
+
+		assertEquals("new_access_token", decryptedAccessToken);
+		assertEquals(200, response.getStatusCode());
+		assertTrue(responseBodyNode.has("access_token"));
+	}
```
[the remark this answers is not in the document]
Comment (Raj-Shah1) [reply src/test/java/com/salessparrow/api/unit/lib/salesforce/helper/SalesforceOAuthTokenTest.java:86]: Not needed

**Review by AMAN-BARBARIA — APPROVED** [review]
LGTM

**Pull request merged**